src/admin.c

```diff
@@ -1330,6 +1330,83 @@ static bool admin_cmd_enable(PgSocket *admin, const char *arg)
 	return admin_ready(admin, "ENABLE");
 }
 
+
+
+PgSocket *find_socket_in_list(const char *arg, struct StatList *sockets)
+{
+	struct List *item;
+	int j;
+	int string_comp;
+	char buffer[strlen(arg) + 1];
+	PgSocket *socket;
+
+	statlist_for_each(item, sockets) {
+		socket = container_of(item, PgSocket, head);
+		j = snprintf(buffer, strlen(arg) + 1, "%p", socket);
```

Comment:
I think it would be more efficient to parse the pointer value from the input and compare that to address of the PgSocket. Instead of formatting every client pointer and string comparing them.

P.S. I wouldn't care much about this in most cases, but we're calling this function for every client, so a bit of efficiency seems nice.

Reply:
I agree this solution is cleaner. Have a solution that works on all of the linux test cases. Fails on windows for some reason. Don't have a windows dev environment set up for pgbouncer though.

Reply:
Okay I got it working now.